File: python/src/models/donor.py
from sqlalchemy import Column, Integer, String
from sqlalchemy.orm import relationship
from database import Base
from enums import DonorRhEnum


class Donor(Base):
    __tablename__ = "donor"
    id_donor = Column(Integer, primary_key=True, nullable=False, autoincrement=True)
    name = Column(String(30), nullable=False)
    surname = Column(String(30), nullable=False)
    mail = Column(String(30), nullable=False)
    password = Column(String(30), nullable=False)
    blood_type = Column(String(5), nullable=False)
    Rh = Column(Enum(DonorRhEnum), nullable=False)
    donations = relationship("Donation", back_populates="donor")

File: nodejs/database/models/donor.js
'use strict';

module.exports = (sequelize, DataTypes) => {
    const donor = sequelize.define('donor', {
        id_donor: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true
        },
        name: {
            type: DataTypes.STRING(30),
        },
        surname: {
            type: DataTypes.STRING(30),
        },
        mail: {
            type: DataTypes.STRING(30),
        },
        password: {
            type: DataTypes.STRING(30),
        },
        blood_type: {
            type: DataTypes.STRING(5),
        },
        Rh: {
            type: DataTypes.ENUM(['Positive', 'Negative']),
        },
    }, {
        freezeTableName: true,
        timestamps: false,
        underscored: true,
        name: {
            singular: 'donor',
            plural: 'donor'
        }
    });

    donor.associate = models => {
        donor.hasMany(models.donation, {
            foreignKey: 'id_donor',
            onDelete: 'cascade'
        });
    };

    return donor;
};
